### ISSUE TITLE
GlobalEmbeddingSpaceTester documentation

### ISSUE THREAD

**Issue by johnjwatson** [KevinMusgrave/pytorch-metric-learning#182]
I am using `GlobalEmbeddingSpaceTester` but I am unsure of what it actually does :( the docs seem to say it just computes knn, but nothing more. I use it in this way (as per Kevin's suggestion from my earlier post):

accuracy_calculator = AccuracyCalculator(k = 50)
tester = GlobalEmbeddingSpaceTester(accuracy_calculator = accuracy_calculator)
tester.test(dataset_dict, epoch, model)

and I get something like:
{"test": {"epoch": 2, "AMI_level0": 0.5238518671512875, "NMI_level0": 0.9118394595053246, "mean_average_precision_at_r_level0": 0.645671834625323, "precision_at_1_level0": 0.7558139534883721, "r_precision_level0": 0.6531007751937984}}

Unfortunately, I have not seen these AMI/NMI metrics before :(

I am a bit confused about what it is really doing under the hood. A high level explanation would be nice - and hopefully I will pick it up from here :)

**Comment by KevinMusgrave** [comment]
"Global" means that during k-nn, the tester considers all points in the dataset split you've provided. This seems like the obvious thing to do, but there are situations where you might want different assumptions (e.g. [WithSameParentLabelTester](https://kevinmusgrave.github.io/pytorch-metric-learning/testers/#withsameparentlabeltester))

For each of the accuracy metrics, "level0" means the 0th label hierarchy level. This only matters if you're dealing with multi-label datasets.

Here's some info on the various accuracy metrics:

AMI: 
 - https://scikit-learn.org/stable/modules/generated/sklearn.metrics.adjusted_mutual_info_score.html
 - https://en.wikipedia.org/wiki/Adjusted_mutual_information

NMI: 
 - https://scikit-learn.org/stable/modules/generated/sklearn.metrics.normalized_mutual_info_score.html
 - https://course.ccs.neu.edu/cs6140sp15/7_locality_cluster/Assignment-6/NMI.pdf

mean_average_precision_at_r:
 - See section 3.2: https://arxiv.org/pdf/2003.08505.pdf

precision_at_1:
 - Fancy way of saying "is the 1st nearest neighbor correct?"

r_precision:
 - See section 3.2: https://arxiv.org/pdf/2003.08505.pdf

**Comment by KevinMusgrave** [comment]
The information above was added to the documentation.
